bindings/rust/src/lib.rs

```diff
@@ -9,6 +9,33 @@ mod sys;
 use std::ffi::CString;
 use std::ptr::NonNull;
 
+#[derive(Eq, PartialEq, Debug, Clone)]
+pub enum Error2 {
+    ParsingFailure,
```

Comment:
In this PR the main question is whether we should go with error codes like this, or free text.

Reply:
If we have the enum, it is much easier to compare outcome in tests, but we need to add display trait implementations so those codes can be displayed with a text counterpart during printing (such as `println!("{:#}", err)`).

Reply:
I'd got with string first.